### PR TITLE
Add saving of push notification subs into firebase

### PR DESCRIPTION
**Pull request by LoneRifle**
* Register our push notification service worker when our app fires up
* When a user takes a queue number, register (with his permission) his browser
for push notifications. Persist the resulting subscription JSON in Firebase
* Adjust our build files to nicely play with the new listener

* `created_at` -> `createdAt`
* save uid for access control purposes

TODO: As Web Push restricts us to one subscription per user per site, figure a way
to track which queues we have signed up for, and unsubscribe only if we are not
on any more queues

### CODE REVIEW

File: push-listener.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+self.addEventListener('push', function (event) {
+  if (event.data) {
+    const payload = event.data.json()
+    const position = Number(payload.position)
+    const body = position > 0
+      ? `You are now #${position} in the queue. Please be ready to be served.`
```
Comment (joelchoo) [inline push-listener.js]: Is having both `position` and `queueNo` too confusing for the user?

File: push-listener.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      : "It's now your turn! Please be ready to be served."
+    const promiseChain = self.registration.showNotification(payload.queueName, { body })
```
Comment (joelchoo) [inline push-listener.js]: better var name?

File: push-listener.js (hunk 1 above)
Comment (LoneRifle) [reply push-listener.js]: Okay, but I need to figure out how to convey that they are a number of places away from being served without the clumsiness of the sentence i just uttered.

File: src/components/QueueView.vue [hunk 3]
```diff
@@ -65,15 +80,26 @@ export default {
 
       let vue = this
 
-      function complete (err, success, newVal) {
+      async function complete (err, success, newVal) {
```
Comment (joelchoo) [inline src/components/QueueView.vue]: Does changing it to async affect Firebase?

Comment (LoneRifle) [reply src/components/QueueView.vue]: It doesn't. the callback is invoked but inspection of the lib suggests that it fire-and-forgets

File: push-listener.js [hunk 4]
```diff
@@ -3,21 +3,10 @@ self.addEventListener('push', function (event) {
...
-      promiseChain.then(async () => {
-        const {pushManager} = self.registration
-        const oldSubscription = await pushManager.getSubscription()
-        if (oldSubscription) {
-          await oldSubscription.unsubscribe()
-        }
-      })
-    }
-    event.waitUntil(promiseChain)
+    const notifyUser = self.registration.showNotification(payload.queueName, { body })
```
Comment (joelchoo) [inline push-listener.js]: No need to do for now, but maybe showNotification only if position <= some threshold?

**Pull request merged**